object-file.c

```diff
@@ -2027,7 +2027,7 @@ int write_object_file_flags(const void *buf, unsigned long len,
 	return write_loose_object(oid, hdr, hdrlen, buf, len, 0, flags);
 }
 
-int hash_object_file_literally(const void *buf, unsigned long len,
+int hash_object_file_literally(const void *buf, size_t len,
```

Comment:
Not sure what to do with this (far too much in the one comment). 

'We' needed to start somewhere. 

The first commit identified (and created) a failing test that has the tightest of focus. The changes then work from tat specific failure, working outward, until we have a passing test, then we can flip the test status. It only resolves that very narrowest of tests, leaving the rest still untested and failing (the future changes). 

One factor this series did drive out (for me) is that the hash functions also needed updating, not just the zlib functions and the many internal Git length type usages.

Reply:
> Not sure what to do with this (far too much in the one comment).

You need to practice writing informative commit messages.

> 'We' needed to start somewhere.

Sure. And we now need to tell a story with the patch series. And like every good story, it needs to be edited. Nobody cares how we arrived at the end, as long as the story is fun to read.

> The first commit identified (and created) a failing test that has the tightest of focus.

Yes, that's how you developed the patch series. And the last patch in the patch series changes something in the code that lets the test case pass. But it does not have to end there. The first commit can still stay in place, but the last commit should probably be put earlier in the patch series. Necessarily, the change from `test_expect_failure` to `test_expect_success` will have to be moved to the patch that will now be at the end.

Reply:
> > Not sure what to do with this (far too much in the one comment).
> 
> You need to practice writing informative commit messages.

At my day job, this problem came up, and the best advice that came out of the session was: pretend that you are talking your future self, like, 6 months from now, when you don't recall _any_ details. What would you tell your future self to catch them up?

I am usually loathe to suggest commit messages to native English speakers (because English is only a second language to me, or third, depending how you want to count). So I will need encouragement from your side if you want me to.

Reply:
I've updated the commit messages, but retained the 'find, fix, find again' processing, and the half-stepping (splitting at signature, then do internals), though all false steps have been rebased away.

I have also added three additional tests at the end, because they passed (yay) which gives a bit of confidence there weren't other gotcha's in those other parallel paths.

Reply:
I haven't updated the PR cover note yet.